src/test/java/com/google/sps/CommentRetrievalTest.java

```diff
@@ -0,0 +1,138 @@
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.mock;
+
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.model.CommentThread;
+import com.google.api.services.youtube.model.CommentThreadListResponse;
+import com.google.sps.servlets.utils.YouTubeCommentRetriever;
+import java.util.Collections;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+// Using wildcard import since documentation is unclear which package
+// "USES_DEEP_STUBS" is imported from
+
+@RunWith(JUnit4.class)
+public class CommentRetrievalTest {
+
+  private final int HUNDRED = 100;
+  private final int EXCESS_HUNDRED = 200;
+  private final int MAX_COMMENTS_PER_TOKEN = 100;
+  private YouTubeCommentRetriever commentRetriever;
+  private final String videoUrl = "dummyVideo";
+  private final String NEXT_PAGE_TOKEN = "Some Page Token";
+  private final String FIRST_TOKEN_COMMENT = "First";
+  private final String SECOND_TOKEN_COMMENT = "Second";
+
+  // Simulate a CommentThreadListResponse with exactly as many comments as expected
+  public void setUpYouTubeMocks(int numExpectedComments) throws Exception {
+    YouTube mockedYoutube = mock(YouTube.class, RETURNS_DEEP_STUBS);
+    YouTube.CommentThreads.List mockedCommentThreadList =
+        mock(YouTube.CommentThreads.List.class, RETURNS_DEEP_STUBS);
+    boolean useNextPageToken = (numExpectedComments > MAX_COMMENTS_PER_TOKEN);
+    // This is only enough for two calls, all these tests require
+    // Return on the first call either 100 comments or the desired number.
+    // If there's no next page token, return nothing on the second call,
+    // Otherwise return the remainder of comments
+    CommentThreadListResponse firstResponse =
+        mockThreadListResponse(
+            Math.min(numExpectedComments, MAX_COMMENTS_PER_TOKEN),
+            (useNextPageToken) ? NEXT_PAGE_TOKEN : null,
+            FIRST_TOKEN_COMMENT);
+    if (useNextPageToken) {
+      CommentThreadListResponse secondResponse =
+          mockThreadListResponse(
+              Math.min(numExpectedComments - MAX_COMMENTS_PER_TOKEN, MAX_COMMENTS_PER_TOKEN),
+              NEXT_PAGE_TOKEN,
+              SECOND_TOKEN_COMMENT);
+      when(mockedCommentThreadList.execute()).thenReturn(firstResponse, secondResponse);
+    } else {
+      when(mockedCommentThreadList.execute()).thenReturn(firstResponse);
+    }
+
+    when(mockedYoutube.commentThreads().list(anyString())).thenReturn(mockedCommentThreadList);
+    commentRetriever = new YouTubeCommentRetriever(mockedYoutube);
+  }
+
+  // Creates list of comments of desired length.
+  // Page token is needed for iteration but its contents are irrelevant.
+  private CommentThreadListResponse mockThreadListResponse(
+      int desiredComments, String nextPageToken, String commentContent) {
```

Comment:
rename desiredComments to numDesiredComments to make it more clear that it is a number.